core/src/io/anuke/mindustry/entities/type/Player.java

```diff
@@ -103,6 +104,9 @@ public void hitboxTile(Rectangle rectangle){
     public void onRespawn(Tile tile){
         boostHeat = 1f;
         achievedFlight = true;
+        if(state.rules.limitedRespawns){
```

Comment:
(refers to comment on line 522, 523)

https://github.com/Anuken/Mindustry/blob/b9421f8c5ab6ecdbf1b1ddf43ad592abc1388911/core/src/io/anuke/mindustry/entities/type/Player.java#L107-L109

```suggestion
if(state.rules.respawns == 0){
  respawns--;
}
```